test/functional/indices/nulls-not-distinct/entity/IndexEntity.ts

```diff
@@ -0,0 +1,36 @@
+import { Entity } from "../../../../../src/decorator/entity/Entity"
+import { Column } from "../../../../../src/decorator/columns/Column"
+import { PrimaryGeneratedColumn } from "../../../../../src/decorator/columns/PrimaryGeneratedColumn"
+import { Index } from "../../../../../src"
+
+@Entity()
+@Index("unique_index_nulls_not_distinct", ["value", "other_value"], {
+    unique: true,
+    nullsNotDistinct: true,
+})
+export class IndexEntityNullsNotDistinct {
+    @PrimaryGeneratedColumn()
+    id: number
+
+    @Column({ type: "text", nullable: true })
```

Comment:
We can maybe remove type and let it be inferred from the `string` type of the `value` property.

Then, if other drivers would support this feature, they would only need to add the driver to the list of supported drivers in the test file.